lib/puppetserver/ca/action/infracrl.rb

```diff
@@ -0,0 +1,114 @@
+require 'optparse'
+
+require 'puppetserver/ca/utils/cli_parsing'
+require 'puppetserver/ca/config/puppet'
+require 'puppetserver/ca/errors'
+require 'puppetserver/ca/utils/file_system'
+require 'puppetserver/ca/utils/signing_digest'
+require 'puppetserver/ca/local_certificate_authority'
+
+module Puppetserver
+  module Ca
+    module Action
+      class InfraCRL
+        include Puppetserver::Ca::Utils
+
+        SUMMARY = "Setup infrastructure CRL based on a node inventory."
+        BANNER = <<-BANNER
+Usage:
+  puppetserver ca infracrl [--help]
+  puppetserver ca infracrl [--config PATH]
+
+Description:
+  Creates auxiliary files needed to use the infrastructure-only CRL.
+  Assumes the existence of an `infra_inventory.txt` file in the CA
+  directory listing the certnames of the infrastructure nodes in the
+  Puppet installation. Generates the `infra_serials` file and the empty
+  CRL to be populated with revoked infrastructure nodes.
+
+  To determine the target location, the default puppet.conf
+  is consulted for custom values. If using a custom puppet.conf
+  provide it with the --config flag
+
+Options:
+BANNER
+
+        def initialize(logger)
+          @logger = logger
+        end
+
+        def run(input)
+          # Validate config_path provided
+          config_path = input['config']
+          if config_path
+            errors = FileSystem.validate_file_paths(config_path)
+            return 1 if Errors.handle_with_usage(@logger, errors)
+          end
+
+          puppet = Config::Puppet.new(config_path)
+          puppet.load
+          settings = puppet.settings
+          return 1 if Errors.handle_with_usage(@logger, puppet.errors)
+
+          inventory_file = File.join(settings[:cadir], 'infra_inventory.txt')
+          if !File.exist?(inventory_file)
+            @logger.err "Please create an inventory file at '#{inventory_file}' with
+              the certnames of your infrastructure nodes before proceeding with
+              infra CRL setup!"
+            return 1
+          end
+
+          # This can be left blank, puppetserver will populate it with the serials for
+          # the certnames in the inventory file when the server starts
+          FileSystem.write_file(File.join(settings[:cadir], 'infra_serials'), '', 0644)
```

Comment:
Should we checking if this exists first?

Reply:
Oh, and not creating one if it does? Yeah probably.